scripts/generate_embeddings_openstack.py

```diff
@@ -33,37 +33,153 @@ def url_function(self, file_path):
+        )
+
+
+def get_argument_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(description="Generate vector DB embeddings!")
+
+    # TODO(lpiwowar): We should be able to support building of a single vector database
+    #                 for both upstream and downstream documentation
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "-rf",
+        "--rhoso-folder",
+        type=Path,
+        required=False,
+        help="Directory containing the plain text RHOSO documentation",
+    )
+    group.add_argument(
+        "-of",
+        "--openstack-folder",
+        type=Path,
+        required=False,
+        help="Directory containing the plain text OpenStack documentation",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        required=True,
+        type=Path,
+        help="Vector DB output folder",
+    )
+    parser.add_argument(
+        "-i",
+        "--index",
+        required=True,
+        type=str,
+        help="Index name that should be attached to the data stored in the vector DB",
+    )
+    parser.add_argument(
+        "-w",
+        "--workers",
+        required=True,
+        default=-1,
+        type=int,
+        help=(
+            "Number of workers to parallelize the data loading. Set to a "
+            "negative value by default, turning parallelism off"
+        ),
+    )
+    parser.add_argument(
+        "-md",
+        "--model-dir",
+        required=True,
+        type=Path,
+        default="embeddings_model",
+        help="Directory containing the embedding model",
+    )
+    parser.add_argument(
+        "-mn",
+        "--model-name",
+        required=True,
+        type=str,
+        help="HF repo id of the embedding model",
+    )
+    parser.add_argument(
+        "-c",
+        "--chunk",
+        required=False,
+        type=int,
+        default=380,
+        help="Chunk size for embedding",
+    )
+    parser.add_argument(
+        "-l",
+        "--overlap",
+        required=False,
+        type=int,
+        default=0,
+        help="Chunk overlap for embedding",
+    )
+    parser.add_argument(
+        "-em",
+        "--exclude-metadata",
+        nargs="+",
+        required=False,
+        type=str,
+        default=None,
+        help="Metadata to be excluded during embedding",
+    )
+    # TODO(lpiwowar): Add support for different vector stores
+    parser.add_argument(
+        "--vector-store-type",
+        default="faiss",
+        choices=["faiss"],
+        help="vector store type to be used.",
+    )
+
+    return parser
+
+
 if __name__ == "__main__":
-    parser = utils.get_common_arg_parser()
+    parser = get_argument_parser()
     args = parser.parse_args()
-    print(f"Arguments used: {args}")
 
-    output_dir = os.path.normpath("/" + args.output).lstrip("/")
+    output_dir = os.path.normpath("/" + str(args.output)).lstrip("/")
```

Comment:
Since you already use a Path object, it would make sense to use it everywhere, rather than string. That or keep the current string based approach.

Reply:
Done:)